openedx/core/djangoapps/user_api/accounts/tests/test_retirement_views.py

```diff
@@ -0,0 +1,1492 @@
+        """
+        Create the given number of test users and UserRetirementPartnerReportingStatus rows,
+        enroll them in the given course (or the default test course if none given), and set
+        their processing state to "is_being_processed".
+
+        Returns a list of user dicts representing what we would expect back from the
+        endpoint for the given user / enrollment.
+        """
+        user_dicts = []
+        courses = self.courses if courses is None else courses
+
+        for _ in range(num):
+            user = UserFactory()
+            UserRetirementPartnerReportingStatus.objects.create(
+                user=user,
+                original_username=user.username,
+                original_email=user.email,
+                original_name=user.first_name + ' ' + user.last_name,
+                is_being_processed=is_being_processed
+            )
+
+            enrollments = []
+            for course in courses:
+                enrollments.append(CourseEnrollment.enroll(user=user, course_key=course.id))
+
+            user_dicts.append(
+                self.get_user_dict(user, enrollments)
+            )
+
+        return user_dicts
+
+    def assert_status_and_user_list(self, expected_users, expected_status=status.HTTP_200_OK):
+        """
+        Makes the partner reporting list GET and asserts that the given users are
+        in the returned list, as well as asserting the expected HTTP status code
+        is returned.
+        """
+        response = self.client.post(self.url, **self.headers)
+        self.assertEqual(response.status_code, expected_status)
+
+        returned_users = response.json()
+        print(returned_users)
+        print(expected_users)
+
+        self.assertEqual(len(expected_users), len(returned_users))
+
+        # Due to the nested list of orgs we can't use assertCountEqual for the whole thing,
+        # this works around that
+        found = False
+        for testing_user in expected_users:
+            for returned_user in returned_users:
+                if returned_user['original_username'] == testing_user['original_username']:
+                    found = True
+                    # Check the orgs sub-list
+                    self.assertCountEqual(returned_user['orgs'], testing_user['orgs'])
+
+                    # Remove the orgs so we can compare the rest of the structure
+                    del returned_user['orgs']
+                    del testing_user['orgs']
+                    self.assertCountEqual(returned_user, testing_user)
+
+                    # Remove this user from the list, when we're done it should be empty
+                    returned_users.remove(returned_user)
+                    break
+            # Check that each expected user was found in the returned list
+            self.assertTrue(found)
+        # Check that there are no extra users returned that weren't expected
+        self.assertFalse(returned_users)
```

Comment:
Perhaps just sort both lists and compare sequentially here? But perhaps not worth the effort.

Reply:
That's a better solution than this kludge, I'll give that a shot. 👍